src/scanner/past.rs

```diff
@@ -26,29 +26,32 @@ pub async fn scan<'a>(
         context.chain_id,
     )
     .await?;
+
     let block_number = signer
         .get_block_number()
         .await
-        .context(format!(
-            "could not get current block number for chain {}",
-            context.chain_id
-        ))?
+        .context("could not get current block number")?
         .as_u64();
 
     let mut db_connection = context
         .db_connection_pool
         .clone()
         .get()
         .context("could not get database connection")?;
-    let snapshot_block =
-        match models::Checkpoint::get_for_chain_id(&mut db_connection, context.chain_id) {
-            Ok(snapshot_block) => Some(u64::try_from(snapshot_block.block_number).unwrap()),
-            Err(error) => {
-                tracing::error!("could not get snapshot block - {:#}", error);
-                None
-            }
-        };
-    let mut from_block = snapshot_block.unwrap_or(context.factory_config.deployment_block);
+    let checkpoint_block =
+        models::Checkpoint::get_for_chain_id(&mut db_connection, context.chain_id)
+            .context("could not get checkpoint block")?
+            .map(|checkpoint| {
+                // realistically, the following should never happen
+                u64::try_from(checkpoint.block_number).expect(
+                    format!(
+                        "could not convert checkpoint block number {} to usigned integer",
```

Comment:
Small type here `usigned`.

Reply:
Addressed in 5e71d3f3f6bc58bc623f5004acd7070f77955ac1